vlm/src/main/scala/geotrellis/contrib/vlm/RasterSummary.scala

```diff
@@ -0,0 +1,100 @@
+import org.apache.spark.rdd.RDD
+
+case class RasterSummary(
+  crs: CRS,
+  cellType: CellType,
+  cellSize: CellSize,
+  extent: Extent,
+  cells: Long,
+  count: Long
+) {
+
+  def estimatePartitionsNumber: Int = {
+    import squants.information._
+    val bytes = Bytes(cellType.bytes * cells)
+    val numPartitions: Int = math.max((bytes / Megabytes(64)).toInt, 1)
+    println(s"Using $numPartitions partitions for ${bytes.toString(Gigabytes)}")
+    numPartitions
+  }
+
+  def levelFor(layoutScheme: LayoutScheme): LayoutLevel =
+    layoutScheme.levelFor(extent, cellSize)
+
+  def toRasterExtent: RasterExtent = RasterExtent(extent, cellSize)
+
+  def layoutDefinition(scheme: LayoutScheme): LayoutDefinition = scheme.levelFor(extent, cellSize).layout
+
+  def combine(other: RasterSummary): RasterSummary = {
+    require(other.crs == crs, s"Can't combine LayerExtent for different CRS: $crs, ${other.crs}")
+    val smallestCellSize = if (cellSize.resolution < other.cellSize.resolution) cellSize else other.cellSize
+    RasterSummary(
+      crs,
+      cellType.union(other.cellType),
+      smallestCellSize,
+      extent.combine(other.extent),
+      cells + other.cells,
+      count + other.count
+    )
+  }
+
+  def toTileLayerMetadata(layoutLevel: LayoutLevel): (TileLayerMetadata[SpatialKey], Some[Int]) = {
+    val LayoutLevel(zoom, layoutDefinition) = layoutLevel
+    val re = TargetGrid(layoutDefinition)(toRasterExtent)
+    val dataBounds = KeyBounds(layoutDefinition.mapTransform.extentToBounds(re.extent))
+    TileLayerMetadata[SpatialKey](cellType, layoutDefinition, re.extent, crs, dataBounds) -> Some(zoom)
+  }
+
+  def toTileLayerMetadata(layoutDefinition: LayoutDefinition, zoom: Int): (TileLayerMetadata[SpatialKey], Some[Int]) =
+    toTileLayerMetadata(LayoutLevel(zoom, layoutDefinition))
+
+  def toTileLayerMetadata(layoutType: LayoutType): (TileLayerMetadata[SpatialKey], Option[Int]) = {
+    val (ld, zoom) = layoutType.layoutDefinitionWithZoom(crs, extent, cellSize)
+    val dataBounds: Bounds[SpatialKey] = KeyBounds(ld.mapTransform.extentToBounds(extent))
+    TileLayerMetadata[SpatialKey](cellType, ld, extent, crs, dataBounds) -> zoom
+  }
+
+  def resample(resampleGrid: ResampleGrid): RasterSummary = {
+    val re = resampleGrid(toRasterExtent)
+    RasterSummary(
+      crs      = crs,
+      cellType = cellType,
+      cellSize = re.cellSize,
+      extent   = re.extent,
+      cells    = re.size,
+      count    = count
+    )
+  }
+}
+
+object RasterSummary {
+  /** Collect [[RasterSummary]] from unstructred rasters, grouped by CRS */
+  def collect[V <: CellGrid: GetComponent[?, ProjectedExtent]](rdd: RDD[V]): Seq[RasterSummary] = {
+    rdd
+      .map { grid =>
+        val ProjectedExtent(extent, crs) = grid.getComponent[ProjectedExtent]
+        val cellSize = CellSize(extent, grid.cols, grid.rows)
+        (crs, RasterSummary(crs, grid.cellType, cellSize, extent, grid.size, 1))
```

Comment:
@echeipesh Did we want the number of non-`NoData` values for this PR? Or can that be pushed off until later?

Reply:
At this point it's not resolved actually :/ I'd like to resolve in a separate PR. At the moment cells is just a `cols x rows` value.

Reply:
That sounds good to me :+1: